### PR TITLE
Issues/520

### LINKED ISSUE #520: New Shortcode attribute: button_only=true
Users often request the ability to have just a "Donate" button that opens the whole form in a modal rather than choosing the amount first. This could be achieved by adding an attribute to the shortcode called `button_only`
## CONSIDERATIONS

This would conflict with the existing attribute display_mode. To compensate for that we should do the following:
- button_only should have a default of false
- If button_only is true it overrides whatever might be in display_mode
- Naturally, the shortcode generator will need to output these correctly, but the generator form itself should disable the display mode option if button_only is enabled.

### PR DESCRIPTION
**Pull request by ravinderk**
## Description

This pr will fix #520 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by ravinderk** [comment]
@DevinWalker @mathetos 
screen shot 2016-11-24 at 10 14 53 am


Currently, form heading appearing on single form page in `h1` tag and it is obviously outside of form wrapper `div`, so that why form title will not show in modal even if the developer calls it with function. I need default form title inside the wrapper to show this in modal, but now on a single page, we have form title in `h1` and `h2` tag. Is there SEO issue or anything related to this?
Do I need to check conditionally that if admin showing form on single page or any other to output form title conditionally and use existing `h1` heading on single form page instead of print `h2` for all page.

**Comment by DevinWalker** [comment]
@ravinderk please do the conditional check so the title isn't output twice. I know we'll get nailed by the SEOs out there about duplicate content.

File: includes/forms/template.php [hunk 1]
```diff
@@ -90,11 +90,17 @@ function give_get_donation_form( $args = array() ) {
...
-			if ( isset( $args['show_title'] ) && $args['show_title'] == true ) {
-
-				echo apply_filters( 'give_form_title', '<h2 class="give-form-title">' . get_the_title( $form_id ) . '</h2>' );
-
+			/**
+			 * Show form title:
+			 * 1. if show_title params set to true
+			 * 2. if admin set form display_style to button
+			 */
+			$form_title = apply_filters( 'give_form_title', '<h2 class="give-form-title">' . get_the_title( $form_id ) . '</h2>' );
```
Comment (DevinWalker) [inline includes/forms/template.php]: Conditional for whether on the single form page or not like you mention

**Comment by mathetos** [comment]
Ya, that's important to avoid. But I'd also recommend that the title is NOT an <h1> whenever it's output via the shortcode. There should always only be ONE <h1> on each page, and the shortcode most often would be used to add the donation form to a page/post that ALREADY has an <h1> This might be a separate issue but it's an important one SEO-wise.

**Pull request merged**